action.yml

```diff
@@ -1,36 +1,119 @@
+    description: "The tutor version matrix to use."
     required: true
   shell_file_to_run:
-    description: 'The path of the shell file to run the integration tests.'
+    description: "The path of the shell file to run the integration tests."
+    required: true
+  openedx_extra_pip_requeriments:
+    description: "Optional extra pip requirements to install in Open edX. E.g: 'package1==1.0 package2>=2.0'"
+    required: false
+    default: ""
+  python_version:
+    description: "The Python version to use for running the tests."
+    required: false
+    default: "3.11"
+  fixtures_file:
+    description: "Optional path to the plugin's fixtures file to load."
     required: false
 
 runs:
-  using: 'composite'
+  using: "composite"
   steps:
-  - name: Prepare Tutor and Launch
-    run: |
-      pip install "tutor$INPUT_TUTOR_VERSION"
-      TUTOR_ROOT="$(pwd)" tutor --version
-      TUTOR_ROOT="$(pwd)" tutor config save
-      TUTOR_ROOT="$(pwd)" tutor mounts add lms,cms,lms-worker,cms-worker:$(pwd)/$INPUT_APP:/openedx/$INPUT_APP
-      chmod 777 . -R
-      TUTOR_ROOT="$(pwd)" tutor local launch -I
-    shell: bash
-    env:
-      INPUT_APP: ${{ inputs.app_name }}
-      INPUT_TUTOR_VERSION: ${{ inputs.tutor_version }}
-
-  - name: Run integration tests in lms
-    if: ${{ inputs.shell_file_to_run }}
-    run: |
-      TUTOR_ROOT="$(pwd)" tutor local run lms bash $INPUT_SHELL_FILE
-    shell: bash
-    env:
-      INPUT_SHELL_FILE: /openedx/${{ inputs.app_name }}/${{ inputs.shell_file_to_run }}
+    - name: Checkout code
+      uses: actions/checkout@v4
+
+    - name: Set up Python
+      uses: actions/setup-python@v5
+      with:
+        python-version: ${{ inputs.python_version }}
+
+    - name: Set Tutor environment variables
+      run: |
+        echo "LMS_HOST=local.edly.io" >> "$GITHUB_ENV"
+        echo "CMS_HOST=studio.local.edly.io" >> "$GITHUB_ENV"
+        echo "TUTOR_ROOT=$(pwd)" >> "$GITHUB_ENV"
+        echo "TUTOR_PLUGINS_ROOT=$(pwd)/plugins/" >> "$GITHUB_ENV"
+      shell: bash
+
+    - name: Install and prepare Tutor
+      run: |
+        pip install "tutor${{ inputs.tutor_version }}"
+        tutor config save --set LMS_HOST=$LMS_HOST --set CMS_HOST=$CMS_HOST
+        chmod 777 . -R
+        tutor local launch -I
+      shell: bash
+
+    - name: Configure Caddyfile and Open edX settings
+      run: |
+        mkdir -p plugins
+        cat << 'EOF' > plugins/patches.yml
+
+        name: patches
+        patches:
+          caddyfile: |
+            {$default_site_port} {
+              import proxy "lms:8000"
+            }
+          openedx-cms-production-settings: |
+            ALLOWED_HOSTS = ["*"]
+            ALLOWED_AUTH_APPLICATIONS = ['cms-sso', 'cms-sso-dev']
+          openedx-lms-production-settings: |
+            ALLOWED_HOSTS = ["*"]
+            ALLOWED_AUTH_APPLICATIONS = ['cms-sso', 'cms-sso-dev']
+        EOF
+        tutor plugins enable patches
```

Comment:
Should we move this file to a Caddyfile outside the action definition, or would that be overkill? Composite actions can access their files, so maybe we can leverage that, although it might be more complex for little gain.

Reply:
Created a patch file that I'm importing in the step